src/main/java/io/vertx/core/http/impl/Http1xServerRequest.java

```diff
@@ -346,15 +331,37 @@ public HttpServerRequest exceptionHandler(Handler<Throwable> handler) {
   @Override
   public HttpServerRequest pause() {
     synchronized (conn) {
-      pendingQueue().pause();
+      if (pending != null) {
+        pending.pause();
+      } else {
+        pending = InboundBuffer.createPaused(context, 8, pendingDrainHandler(), pendingHandler());
+      }
       return this;
     }
   }
 
+  private Handler<Object> pendingHandler() {
```

Comment:
These are to make sure just a single lambda type is generated - If I would inline them I'm not sure they'll be the same lambda type.